### ISSUE TITLE
fail-under should show more precision; confusing when just a small value below threshold

### ISSUE THREAD

**Issue by nedbat** [coveragepy/coveragepy#284]
*Originally reported by* **Jon Dufresne (Bitbucket: [jdufresne](https://bitbucket.org/jdufresne), GitHub: [jdufresne](https://github.com/jdufresne))**

----------------------------------------

If the test coverage is just a small fraction below the `fail-under` value, the report will fail with a non-zero exit status (good and expected) but the report appears as if everything is ok. As an example I received the following report:

```
$ coverage report --fail-under=70
Name                                    Stmts   Miss  Cover   Missing
---------------------------------------------------------------------
...  
---------------------------------------------------------------------
TOTAL                                    1816    551    70%   
$ echo $?
2
```

Doing the math:

```
>>> (1816.0 - 551.0) / 1816.0
0.6965859030837004
```

This is indeed below 70%. However at first glance, `coverage` reports this as 70% coverage (due to rounding I assume) creating confusion for me the user. In the case of failure, I would like to see `report` provide a more precise number for coverage so that the failure is immediately obvious.

----------------------------------------
- Bitbucket: https://bitbucket.org/ned/coveragepy/issue/284

**Comment by nedbat** [comment]
Hmm, it seems odd to me to change the reporting based on whether --fail-under is used or not.  Better I think would be to use the rounded number in the fail-under calculation.

**Comment by nedbat** [comment]
*Original comment by* **Jon Dufresne (Bitbucket: [jdufresne](https://bitbucket.org/jdufresne), GitHub: [jdufresne](https://github.com/jdufresne))**

----------------------------------------


> Hmm, it seems odd to me to change the reporting based on whether --fail-under is used or not. Better I think would be to use the rounded number in the fail-under calculation.

Sure. I think this solution makes sense and would remove the confusion.

**Comment by nedbat** [comment]
*Original comment by* **Alexander Walters (Bitbucket: [sdamon](https://bitbucket.org/sdamon), GitHub: [sdamon](https://github.com/sdamon))**

----------------------------------------

Or just use floor instead of round.  0.9999 would round to 100%, and floor to 99%, and would match the expectations of --fail-under.

**Comment by nedbat** [comment]
*Original comment by* **Jon Dufresne (Bitbucket: [jdufresne](https://bitbucket.org/jdufresne), GitHub: [jdufresne](https://github.com/jdufresne))**

----------------------------------------

> Or just use floor instead of round. 0.9999 would round to 100%, and floor to 99%, and would match the expectations of --fail-under.

I like this idea. I created a pull request:

https://bitbucket.org/ned/coveragepy/pull-request/36/use-mathfloor-instead-of-round-when/diff

**Comment by nedbat** [comment]
Fixed in [0a9cb6a6fc61 (bb)](https://bitbucket.org/ned/coveragepy/commits/0a9cb6a6fc61)
